marbl/marbl.go

```diff
@@ -75,6 +76,7 @@ func (s *Stream) loop() {
 }
 
 func (s *Stream) Close() error {
+	s.closec <- struct{}{}
```

Comment:
It's synchronization. We block on closing the channel until the goroutine has read and processed all the frames, because it won't read this struct out of closec until it returns to loop(). close(), on the next line, doesn't block, and (I believe), doesn't write an empty struct. I actually think that line doesn't do anything at all for us.